src/libsystemd/sd-event/sd-event.c

```
@@ -3142,13 +3144,41 @@ static int process_inotify(sd_event *e) {
         return done;
 }
 
+static bool source_ratelimited(sd_event_source *s) {
+        assert(s);
+
+        if (!s->ratelimit)
+                return false;
+
+        if (s->ratelimit->effective)
+                return true;
+
+        if (ratelimit_below(&s->ratelimit->r))
+                return false;
+
+        /* Let's schedule high priority wake after interval elapsed */
+        (void) sd_event_source_set_time(s->ratelimit->timeout_source, s->ratelimit->r.begin + s->ratelimit->r.interval);
+        (void) sd_event_source_set_priority(s->ratelimit->timeout_source, SD_EVENT_PRIORITY_IMPORTANT);
+        (void) sd_event_source_set_enabled(s->ratelimit->timeout_source, SD_EVENT_ON);
```

we should propagate these errors...